### PR TITLE
Integrate reset subcommand

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

**Comment by tboerger** [comment]
screenshot

This is how the question currently looks like.

**Comment by tboerger** [comment]
The functionality should be fine now, but this change depends on a followup PR against crowbar-core to provide the API for invoking the resets.

**Comment by dirkmueller** [comment]
is there a way to avoid the interactive query with a --yes-i-really-know-what-im-doing?

**Comment by MaximilianMeister** [comment]
> is there a way to avoid the interactive query with a --yes-i-really-know-what-im-doing?

a `--force` option like in the installer subcommand would be good, at least for scripting it would make it much easier. can you add that?

**Comment by aplanas** [comment]
`--force` is more expected for corner situations when a drastic action needs to be done (like reseting even if some services are running), I will vote for `--yes`, `-y`. But this a personal preference, so is not important.

**Comment by tboerger** [comment]
No, currently not. Do we really want to offer that? I think we don't want to offer the option to call it through scripting as this is a hard reset explained in the message.

**Comment by tboerger** [comment]
**Depends on https://github.com/crowbar/crowbar-core/pull/176**

**Comment by tboerger** [comment]
I have added a `--yes` flag now. Please review and give your +1 if you are fine with it.

**Comment by aplanas** [comment]
+1

**Comment by MaximilianMeister** [comment]
:+1: thanks

File: lib/crowbar/client/app/proposal.rb [hunk 1]
```diff
@@ -294,6 +294,66 @@ def commit(barclamp, proposal)
...
         end
+
+        desc "reset BARCLAMP [PROPOSAL]",
+          "Reset a proposal for specific barclamp"
+
+        long_desc <<-LONGDESC
+          `reset BARCLAMP [PROPOSAL]` will try to reset the state
+          for the specified barclamp, if you don't provide a proposal
+          it will take default. If you don't provide a proposal the
+          client will select the default or the first proposal. The
```
Comment (MaximilianMeister) [inline lib/crowbar/client/app/proposal.rb]: there is some duplicated info here. `if you don't provide a proposal...`

**Comment by MaximilianMeister** [comment]
:+1:

**Pull request merged**